src/index.spec.ts

```diff
@@ -104,173 +160,183 @@ describe('EppoJSClient E2E test', () => {
-      `${mockExperimentConfig?.rules[0]?.allocationKey}`,
+      `${mockUfcFlagConfig?.allocations[0].key}`,
     );
   });
 
   it('handles logging exception', () => {
     const mockLogger = td.object<IAssignmentLogger>();
     td.when(mockLogger.logAssignment(td.matchers.anything())).thenThrow(new Error('logging error'));
     td.replace(EppoLocalStorage.prototype, 'get', (key: string) => {
-      if (key !== hashedFlagKey) {
+      if (key !== flagKey) {
         throw new Error('Unexpected key ' + key);
       }
-      return mockExperimentConfig;
+      return mockUfcFlagConfig;
     });
     const subjectAttributes = { foo: 3 };
     globalClient.setLogger(mockLogger);
-    const assignment = globalClient.getAssignment('subject-10', flagKey, subjectAttributes);
-    expect(assignment).toEqual('control');
+    const assignment = globalClient.getStringAssignment(
+      'subject-10',
+      flagKey,
+      '',
+      subjectAttributes,
+    );
+    expect(assignment).toEqual('variant-1');
   });
 
   it('only returns variation if subject matches rules', () => {
     td.replace(EppoLocalStorage.prototype, 'get', (key: string) => {
-      if (key !== hashedFlagKey) {
+      if (key !== flagKey) {
         throw new Error('Unexpected key ' + key);
       }
+
       return {
-        ...mockExperimentConfig,
-        rules: [
+        ...mockUfcFlagConfig,
+        allocations: [
           {
-            allocationKey: 'allocation1',
-            conditions: [
+            ...mockUfcFlagConfig.allocations[0],
+            rules: [
               {
-                operator: md5('GT'),
-                attribute: md5('appVersion'),
-                value: Buffer.from('10', 'utf8').toString('base64'),
+                conditions: [
+                  {
+                    attribute: 'appVersion',
+                    operator: 'GT',
+                    value: 10,
+                  },
+                ],
               },
             ],
           },
         ],
       };
     });
-    let assignment = globalClient.getAssignment('subject-10', flagKey, {
+    let assignment = globalClient.getStringAssignment('subject-10', flagKey, '', {
       appVersion: 9,
     });
-    expect(assignment).toEqual(null);
-    assignment = globalClient.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual(null);
-    assignment = globalClient.getAssignment('subject-10', flagKey, {
+    expect(assignment).toEqual('');
+    assignment = globalClient.getStringAssignment('subject-10', flagKey, '');
+    expect(assignment).toEqual('');
+    assignment = globalClient.getStringAssignment('subject-10', flagKey, '', {
       appVersion: 11,
     });
-    expect(assignment).toEqual('control');
+    expect(assignment).toEqual('variant-1');
   });
 
-  describe('getAssignment', () => {
-    const testData = readAssignmentTestData();
-
-    it.each(testData)(
+  describe('UFC General Test Cases', () => {
+    it.each(readAssignmentTestData())(
```

Comment:
This is the basically the same as the other method with the exception of which config we're using in our moc. I bet we could fold them into the same with something like:
```ts
const unobfuscatedTests = readAssignmentTestData();
const obfuscatedTests = unobfuscatedTests.map((t) => ({obfuscate: true, ...t}));
it.each(unobfuscatedTests.concat(obfuscatedTests))(
```
Then in, the below code, if its obfuscated, we do the obfuscated mocking/init